app/models/manageiq/providers/google/cloud_manager/provision/cloning.rb

```diff
@@ -0,0 +1,41 @@
+module ManageIQ::Providers::Google::CloudManager::Provision::Cloning
+  def do_clone_task_check(clone_task_ref)
+    source.with_provider_connection do |google|
+      instance = google.servers.get(clone_task_ref)
+      return true if instance.sshable?
+      return false
+    end
+  end
+
+  def create_disk
+    source.with_provider_connection do |google|
+      return google.disks.create(
+        :name => dest_name,
+        :size_gb => Integer(root_disk_size),
+        :zone_name => dest_availability_zone.ems_ref,
+        :source_image => source.name)
+    end
+  end
+
+  def prepare_for_clone_task
+    clone_options = super
+
+    clone_options[:name] = dest_name
+    clone_options[:disks] = [create_disk.get_as_boot_disk(true, true)]
+    clone_options[:machine_type] = instance_type.ems_ref
+    clone_options[:zone_name] = dest_availability_zone.ems_ref
+
+    clone_options
+  end
+
+  def log_clone_options(clone_options)
+    # todo(lwander)
+  end
+
+  def start_clone(clone_options)
+    source.with_provider_connection do |google|
+      instance = google.servers.create(clone_options)
+      return instance.name
```

Comment:
If possible I would like to keep this consistent with the OpenStack implementation which returns ID and then does the lookup in the `do_clone_task_check` method.